gu-client/examples/gu-client.rs

```diff
@@ -0,0 +1,1017 @@
+                        Command::Exec {
+                            executable: "./gu-render".into(),
+                            args: Vec::new(),
+                        },
+                        Command::UploadFile {
+                            uri: blob.uri(),
+                            file_path: outf.clone(),
+                            format: ResourceFormat::Raw,
+                        },
+                    ]
+                })
+                .and_then(move |results| {
+                    use std::io::prelude::*;
+                    log::debug!(
+                        "{:?}, results={:?}, output_file={}",
+                        node_id,
+                        results,
+                        output_path.display()
+                    );
+
+                    if results
+                        .iter()
+                        .any(|s| s.starts_with("failed to execute command"))
+                    {
+                        return future::Either::B(future::err(Error::Other(format!(
+                            "{:?}",
+                            results
+                        ))));
+                    }
+
+                    let mut f = match fs::OpenOptions::new()
+                        .create_new(true)
+                        .write(true)
+                        .open(output_path)
+                    {
+                        Ok(f) => f,
+                        Err(err) => return future::Either::B(future::err(err.into())),
+                    };
+
+                    future::Either::A(result_blob.download().for_each(move |b| {
+                        f.write_all(b.as_ref())?;
+                        Ok(())
+                    }))
+                })
+                .and_then(move |_| {
+                    logger.do_send(MainSteps::Frame(frame));
+                    Ok(())
+                })
+        })
+        .then(move |r| match r {
+            Ok(v) => Ok(log::debug!("done: {:?}: {:?}", v, worker_id)),
+            Err(e) => Ok(log::error!("err: {:?}: {:?}", e, worker_id)),
+        })
+}
+
+struct TarBuildHelper<W: std::io::Write> {
+    b: tar::Builder<W>,
+    dirs: BTreeSet<PathBuf>,
+}
+
+impl<W: std::io::Write> TarBuildHelper<W> {
+    fn new(b: tar::Builder<W>) -> Self {
+        let mut dirs = BTreeSet::new();
+
+        dirs.insert(PathBuf::new());
+
+        TarBuildHelper { b, dirs }
+    }
+
+    fn add_dir(&mut self, d: &Path) -> io::Result<()> {
+        use tar::{EntryType, Header};
+
+        log::debug!("dir={:?}", d);
+        if self.dirs.contains(d) {
+            return Ok(());
+        }
+        if let Some(parent) = d.parent() {
+            self.add_dir(parent)?;
+        } else {
+            return Ok(());
+        }
+
+        self.dirs.insert(d.to_owned());
+        let mut h = Header::new_ustar();
+        h.set_entry_type(EntryType::Directory);
+        h.set_mode(0o644);
+        h.set_uid(0);
+        h.set_gid(0);
+        h.set_mtime(0);
```

Comment:
Is time deliberately set to 0 (it means 0 seconds after the beginning of 1970 )?

Reply:
hmm, this is default values for new header. maybe that's why Download in dockerman have strange dates.
changed to: 
```        h.set_mtime(
            d.metadata()?
                .modified()?
                .duration_since(std::time::UNIX_EPOCH)
                .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?
                .as_secs(),
        )
```